Expose ingest storage config doc and changelog (#8503)
grafana/mimir: docs/sources/mimir/configure/configuration-parameters/index.md

# The frontend block configures the query-frontend.
[frontend: <frontend>]

ingest_storage:
  # True to enable the ingestion via object storage.
  # CLI flag: -ingest-storage.enabled
  [enabled: <boolean> | default = false]

  kafka:
    # The Kafka backend address.
    # CLI flag: -ingest-storage.kafka.address
    [address: <string> | default = ""]

    # The Kafka topic name.
    # CLI flag: -ingest-storage.kafka.topic
    [topic: <string> | default = ""]

    # The Kafka client ID.
    # CLI flag: -ingest-storage.kafka.client-id
    [client_id: <string> | default = ""]

    # The maximum time allowed to open a connection to a Kafka broker.
    # CLI flag: -ingest-storage.kafka.dial-timeout
    [dial_timeout: <duration> | default = 2s]

    # How long to wait for an incoming write request to be successfully
    # committed to the Kafka backend.
    # CLI flag: -ingest-storage.kafka.write-timeout
    [write_timeout: <duration> | default = 10s]

    # The number of Kafka clients used by producers. When the configured number
    # of clients is greater than 1, partitions are sharded among Kafka clients.

>     # An higher number of clients may provide higher write throughput at the

Reviewer: ```suggestion
    # A higher number of clients may provide higher write throughput at the
```

    # cost of additional Metadata requests pressure to Kafka.
    # CLI flag: -ingest-storage.kafka.write-clients
    [write_clients: <int> | default = 1]

    # The consumer group used by the consumer to track the last consumed offset.
    # The consumer group must be different for each ingester. If the configured

>     # consumer group contains the '<partition>' placeholder, it will be replaced

Reviewer: ```suggestion
    # consumer group contains the '<partition>' placeholder, it is replaced
```

    # with the actual partition ID owned by the ingester. When empty

>     # (recommended), Mimir will use the ingester instance ID to guarantee

Reviewer: ```suggestion
    # (recommended), Mimir uses the ingester instance ID to guarantee
```

    # Milliseconds timestamp after which the consumption of the partition starts
    # at startup. Only applies when consume-from-position-at-startup is
    # timestamp
    # CLI flag: -ingest-storage.kafka.consume-from-timestamp-at-startup
    [consume_from_timestamp_at_startup: <int> | default = 0]

    # The maximum tolerated lag before a consumer is considered to have caught
    # up reading from a partition at startup, becomes ACTIVE in the hash ring
    # and passes the readiness check. Set 0 to disable waiting for maximum
    # consumer lag being honored at startup.
    # CLI flag: -ingest-storage.kafka.max-consumer-lag-at-startup
    [max_consumer_lag_at_startup: <duration> | default = 15s]

    # Enable auto-creation of Kafka topic if it doesn't exist.
    # CLI flag: -ingest-storage.kafka.auto-create-topic-enabled
    [auto_create_topic_enabled: <boolean> | default = true]

    # When auto-creation of Kafka topic is enabled and this value is positive,
    # Kafka's num.partitions configuration option is set on Kafka brokers with
    # this value when Mimir component that uses Kafka starts. This configuration

>     # option specifies the default number of partitions that Kafka broker will
>     # use for auto-created topics. Note that this is Kafka-cluster wide setting,

Reviewer: ```suggestion
    # option specifies the default number of partitions that the Kafka broker
    # uses for auto-created topics. Note that this is a Kafka-cluster wide setting,
```

>     # and applies to any auto-created topic. If setting of num.partitions fails,

Reviewer: ```suggestion
    # and applies to any auto-created topic. If the setting of num.partitions fails,
```

>     # Mimir will proceed anyway, but auto-created topic may have incorrect
>     # number of partitions.

Reviewer: ```suggestion
    # Mimir proceeds anyways, but auto-created topics could have an incorrect
    # number of partitions.
```

    # CLI flag: -ingest-storage.kafka.auto-create-topic-default-partitions
    [auto_create_topic_default_partitions: <int> | default = 0]

    # The maximum size of a Kafka record data that should be generated by the

>     # producer. An incoming write request bigger than this size is split into

Reviewer: ```suggestion
    # producer. An incoming write request larger than this size is split into
```

    # multiple Kafka records. We strongly recommend to not change this setting
    # unless for testing purposes.
    # CLI flag: -ingest-storage.kafka.producer-max-record-size-bytes
    [producer_max_record_size_bytes: <int> | default = 15983616]

  migration:

>     # When both this option and ingest storage is enabled, distributors will
>     # both write to Kafka and ingesters. A write request will be considered
>     # successful only when written to both backends.

Reviewer: ```suggestion
    # When both this option and ingest storage are enabled, distributors
    # write to both Kafka and ingesters. A write request is considered
    # successful only when written to both backends.
```

  # CLI flag: -ingester.ring.spread-minimizing-zones
  [spread_minimizing_zones: <string> | default = ""]

partition_ring:
  # The key-value store used to share the hash ring across multiple instances.

>   # This option needs be set on ingesters, distributors, queriers and rulers

Reviewer: ```suggestion
  # This option needs be set on ingesters, distributors, queriers, and rulers
```

    # The CLI flags prefix for this block configuration is:
    # ingester.partition-ring
    [consul: <consul>]

    # The etcd block configures the etcd client.
    # The CLI flags prefix for this block configuration is:
    # ingester.partition-ring
    [etcd: <etcd>]

    multi:
      # (advanced) Primary backend storage used by multi-client.
      # CLI flag: -ingester.partition-ring.multi.primary
      [primary: <string> | default = ""]

      # (advanced) Secondary backend storage used by multi-client.
      # CLI flag: -ingester.partition-ring.multi.secondary
      [secondary: <string> | default = ""]

      # (advanced) Mirror writes to secondary store.
      # CLI flag: -ingester.partition-ring.multi.mirror-enabled
      [mirror_enabled: <boolean> | default = false]

      # (advanced) Timeout for storing value to secondary store.
      # CLI flag: -ingester.partition-ring.multi.mirror-timeout
      [mirror_timeout: <duration> | default = 2s]

  # Minimum number of owners to wait before a PENDING partition gets switched to
  # ACTIVE.
  # CLI flag: -ingester.partition-ring.min-partition-owners-count
  [min_partition_owners_count: <int> | default = 1]

>   # How long the minimum number of owners should have been enforced before a

Reviewer: ```suggestion
  # How long the minimum number of owners are enforced before a
```

  # PENDING partition gets switched to ACTIVE.
  # CLI flag: -ingester.partition-ring.min-partition-owners-duration
  [min_partition_owners_duration: <duration> | default = 10s]

  # How long to wait before an INACTIVE partition is eligible for deletion. The

>   # partition will be deleted only if it has been in INACTIVE state for at least

Reviewer: ```suggestion
  # partition is deleted only if it has been in INACTIVE state for at least
```

# through OTLP.
# CLI flag: -distributor.otel-metric-suffixes-enabled
[otel_metric_suffixes_enabled: <boolean> | default = false]

> # (experimental) The default consistency level to enforce to queries when using

Reviewer: ```suggestion
# (experimental) The default consistency level to enforce for queries when using
```
